src/java.base/linux/classes/sun/nio/fs/LinuxWatchService.java

```diff
@@ -226,14 +226,11 @@ Object implRegister(Path obj,
             }
 
             // no modifiers supported at this time
-            for (WatchEvent.Modifier modifier : modifiers) {
+            if (modifiers != null) {
```

Comment:
```suggestion
            if (modifiers.length > 0) {
```
Same thing in PollingWatchService.